examples/social-media/dictionary.asyncapi.yaml

```diff
@@ -0,0 +1,59 @@
+asyncapi: 2.2.0
+
+info:
+  title: Dictionary for organization-wide definitions
```

Comment:
This one is super confusing for me 😄 
Kinda reverse the "official" recommendations we always had that align with the spec where `app` is in the center.
I would never advise users to do it this way. I know people do it. I was involved in PRs where we had to make it possible to read info from `components` that is not referenced in `channels`. I would advise users to store schemas as separate files, not related to asyncapi and use `$ref` to them. 
And yeah, it is not addressed in the readme 😄 

Nothing is black or white, I know, but I have a feeling that in official example it should be clearly described in the readme, why it is not this way or another event though the spec say this and that

Reply:
Hmm... 

I see your point 😆 However, for simplicity, having everything in one file, just makes it easier. I could change the file to not follow AsyncAPI structure?


`dictionary.asyncapi.yaml` -> `dictionary.yaml`

```yaml
components:
  messages:
    commentLiked: 
      payload: 
        $ref: '#/components/schemas/commentLikedPayload'
    likeComment: 
      payload: 
        $ref: '#/components/schemas/likeCommentPayload'
    commentChanged: 
      payload: 
        $ref: '#/components/schemas/commentChangedPayload'
    updateCommentLikes: 
      payload: 
        $ref: '#/components/schemas/updateCommentLikesPayload'
  schemas: 
    commentLikedPayload: 
      type: object
      title: commentLikedPayload
      additionalProperties: false
      properties:
        commentId: 
          allOf: 
            - $ref: '#/components/schemas/commentId'
            - description: Id of the comment that was liked
    likeCommentPayload: 
      type: object
      title: likeCommentPayload
      additionalProperties: false
      properties:
        commentId: 
          allOf: 
            - $ref: '#/components/schemas/commentId'
            - description: Id of the comment that should be liked
    commentChangedPayload: 
      type: object
      title: commentChangedPayload
      additionalProperties: false
      properties:
        commentId: 
          allOf: 
            - $ref: '#/components/schemas/commentId'
            - description: Id of the comment that was changed, such as when someone liked it.
    updateCommentLikesPayload:
      type: object
      title: updateCommentLikesPayload
      additionalProperties: false
      properties:
        commentId: 
          allOf: 
            - $ref: '#/components/schemas/commentId'
            - description: Id of the comment that was changed, such as when someone liked it.
    commentId:
      type: string

```

Reply:
still don't get the name like `dictionary` but as long as it is not `asyncapi` file, that is fine. It only still follows `asyncapi` naming and this can confuse, how about 2 files, one called `shared-messages.yaml` and the other `shared-schemas.yaml` and you do not need `components.messages` and stuff like that

Reply:
You dont? 🤔 I thought it would be pretty clear of a file where you lookup definitions, i.e. looking up words in a dictionary to find their meaning 😄 

I see your point... Not sure what would make the most sense for others 🤔

Reply:
for me dictionary is dictionary 😄 notice that when people talk about schema registry solutions they do not call them schema dictionaries. Just my point of view

Reply:
@derberg do you mind writing your comments in https://github.com/asyncapi/spec/issues/628#issuecomment-968050476 🙂

Reply:
I heard the word `catalog` in the past as an alternative somewhere. I think I agree @derberg the name `dictionary` is not the best either.

Reply:
It has been changed now to two separate files, and the dictionary has been removed.